pandas/core/generic.py

```diff
@@ -7352,6 +7352,14 @@ def _clip_with_one_bound(self, threshold, method, axis, inplace):
                 threshold = self._constructor(threshold, index=self.index)
             else:
                 threshold = align_method_FRAME(self, threshold, axis, flex=None)[1]
+
+        # GH 40420
```

Comment:
um, can you not `subset = subset.fillna(np.inf)` (or -np.inf if the other way)?

Reply:
Yep, not for subset, but for the thresholds.